packages/devtools_app/lib/src/standalone_ui/vs_code/devtools/extensions_view.dart

```diff
@@ -113,7 +113,14 @@ class _SidebarDevToolsExtensionsState extends State<SidebarDevToolsExtensions>
               ),
             );
             unawaited(
-              widget.editor.openDevToolsPage(null, page: ext.screenId),
+              widget.editor.openDevToolsPage(
+                null,
+                page: ext.screenId,
+                requiresDebugSession: ext.requiresConnection,
+                // TODO(https://github.com/flutter/devtools/issues/7955): set
+                // this value based on support matrix declared by the extension.
+                prefersDebugSession: true,
```

Comment:
I worry that putting `true` here will also not be ideal, because if there is >1 debug session this will trigger a prompt (and for static tools that will look odd).

If we do the workaround we discussed in Dart-Code, should we leave this until the correct value can be supplied?

Reply:
If this is left null instead of passing true, what will be the behavior for an extension that does not require a connection, but prefers one if one is available?

Reply:
`prefersDebugSession` defaults to:

- `false` if we know it is a static page (eg. it's a 1p page like Deep Links)
- `true` otherwise

This means for an extension, passing `true` and `null` are basically the same. This means either we will not pass URIs to pages that want them, or we prompt for a session (when >1) and pass URIs to pages that don't want them. The purpose of this flag is to avoid having to pick between one of these not-ideal situations.



The current issue is actually with `requiresDebugSession` (I assumed incorrectly earlier), in that it defaults to:

- `false` if we do not have a page ID
- `false` if we page a `pageId` and we _know_ it is a static page (eg. Deep Links)
- `true` otherwise

So the request for the extension is falling into the last case and requiring a debug session (and since we don't have one, we cannot load anything).

The Dart-Code workaround will be that if both values are null and the page is `_ext`, then overwrite the defaults to become:

- `requiresDebugSession`: `false`
- `prefersDebugSession`: `true`

This means we will always try to give a debug session (prompting if > 1), but if there are none, we will continue to launch without it.